Sources/ApolloCodegenLib/FileGenerators/OperationManifestFileGenerator.swift

```diff
@@ -12,9 +12,9 @@ struct OperationManifestItem {
     self.identifier = operation.operationIdentifier
     self.type = operation.definition.operationType
 
-    var source = operation.definition.source
+    var source = operation.definition.source.convertedToSingleLine()
     for fragment in operation.referencedFragments {
-      source += "\n\(fragment.definition.source)"
+      source += "\\n\(fragment.definition.source.convertedToSingleLine())"
```

Comment:
I don't know swift but that doubling of the backslash on the first line looks suspect to me — unless the `source` field is being used in a different way now? That said the tests don't seem to have changed in an odd way...

Reply:
So the `\\n` will actually insert `\n` as characters in the string vs creating a newline. My understanding from discussions with @AnthonyMDev was that we want everything to be single line now, but we should still have a newline character before each fragment, so this change allows the overall string to be single line while still containing the newline characters before each fragment. If this isn't what we intend I'm happy to rework it, this is just what I gathered we wanted.

Reply:
This particular change goes with Tests/ApolloCodegenTests/CodeGeneration/FileGenerators/OperationManifestFileGeneratorTests.swift right? I think it would be interesting to see how a test case that includes a fragment looks before and after the change in this PR. My suspicion is that you will be adding way too many backslashes. But I might be wrong!

Reply:
I guess my point is that it looks like there's a semantic change here so either:

(a) There's some other change in your PR that cancels it out and there's no end-to-end-change
(b) The old code was actually wrong and this is a bug fix
(c) The old code was right and this is a bug

I don't have enough context to know which is which though! (I don't even know what this particular file is used for, honestly.)

Reply:
Ok looking into everything is actually correct, the expected test results do contain a `\\n` because of the need to escape the `\` so when it gets compared to the actual output which only contains `\n` it succeeds as expected.

So to your point, this (a) where it is a specific change in this PR because everything is now supposed to be single line, the original `\n` would add a newline to the definition for each fragment, and the new setup with `\\n` inserts the actual characters `\n` into the definition string before each fragment.

So a sample out in a manfiest file from this code looks like this:
```
{
  "format": "apollo-persisted-query-manifest",
  "version": 1,
  "operations": [
    {
      "id": "4de9229ebd29faf64c0f3f88e61a77b23d7276fc6d24263b1e0378229ce29744",
      "body": "query GetThing { thing { __typename ...Name } }\nfragment Name on Thing { __typename name }",
      "name": "GetThing",
      "type": "query"
    },
  ]
}
```

Reply:
Huh. But a raw newline isn't actually legal JSON — if you're saying the old code did something more like

```
Ok looking into everything is actually correct, the expected test results do contain a \\n because of the need to escape the \ so when it gets compared to the actual output which only contains \n it succeeds as expected.

So to your point, this (a) where it is a specific change in this PR because everything is now supposed to be single line, the original \n would add a newline to the definition for each fragment, and the new setup with \\n inserts the actual characters \n into the definition string before each fragment.

So a sample out in a manfiest file from this code looks like this:

{
  "format": "apollo-persisted-query-manifest",
  "version": 1,
  "operations": [
    {
      "id": "4de9229ebd29faf64c0f3f88e61a77b23d7276fc6d24263b1e0378229ce29744",
      "body": "query GetThing { thing { __typename ...Name } }
fragment Name on Thing { __typename name }",
      "name": "GetThing",
      "type": "query"
    },
  ]
}```

then was it just broken?

Reply:
That's a good point, technically for it to also be valid in the JSON it needs to be escaped as well so in the final JSON it should be `\\n` so the Swift side needs updated to actually put a `\\n` into the JSON string

As for how it used to work I'm not sure will have to go back and look at how things worked originally for APQs.